src/coreclr/jit/codegenxarch.cpp

```diff
@@ -6804,33 +6802,51 @@ void CodeGen::genIntToIntCast(GenTreeCast* cast)
     switch (desc.ExtendKind())
     {
         case GenIntCastDesc::ZERO_EXTEND_SMALL_INT:
+        case GenIntCastDesc::LOAD_ZERO_EXTEND_SMALL_INT:
             ins     = INS_movzx;
             insSize = desc.ExtendSrcSize();
             break;
         case GenIntCastDesc::SIGN_EXTEND_SMALL_INT:
+        case GenIntCastDesc::LOAD_SIGN_EXTEND_SMALL_INT:
             ins     = INS_movsx;
             insSize = desc.ExtendSrcSize();
             break;
 #ifdef TARGET_64BIT
         case GenIntCastDesc::ZERO_EXTEND_INT:
+        case GenIntCastDesc::LOAD_ZERO_EXTEND_INT:
```

Comment:
Are there going to be any issues with `Load` variants and `canSkip` or will this always be "false" because `srcReg` is `REG_NA` or something else?

Reply:
> will this always be "false" because srcReg is REG_NA or something else?

That's right, because of this code in `AreUpper32BitsZero`:
```cpp
// Bail if not writing to the right register
if (id->idReg1() != reg)
{
    return false;
}
```
Additionally, `canSkip` is not used on the "load" path so it does not really matter.